src/controllers/dynakube/activegate/internal/capability/reconciler_test.go

```diff
@@ -1,517 +1,213 @@
-// 		t.Run(tt.name, func(t *testing.T) {
-// 			r := &Reconciler{
-// 				dynakube:   tt.fields.Instance,
-// 				Capability: tt.fields.Capability,
-// 			}
-// 			if got := r.calculateStatefulSetName(); got != tt.want {
-// 				t.Errorf("Reconciler.calculateStatefulSetName() = %v, want %v", got, tt.want)
-// 			}
-// 		})
-// 	}
-// }
-
-// func TestGetContainerByName(t *testing.T) {
-// 	verify := func(t *testing.T, containers []corev1.Container, lookingForContainer string, errorMessage string) {
-// 		container, err := getContainerByName(containers, lookingForContainer)
-// 		if errorMessage == "" {
-// 			assert.NoError(t, err)
-// 			assert.NotNil(t, container)
-// 			assert.Equal(t, lookingForContainer, container.Name)
-// 		} else {
-// 			assert.Error(t, err)
-// 			assert.Contains(t, err.Error(), errorMessage)
-// 			assert.Nil(t, container)
-// 		}
-// 	}
-
-// 	t.Run("empty slice test cases", func(t *testing.T) {
-// 		verify(t, nil, "", `Cannot find container "" in the provided slice (len 0)`)
-// 		verify(t, []corev1.Container{}, "", `Cannot find container "" in the provided slice (len 0)`)
-// 		verify(t, []corev1.Container{}, "something", `Cannot find container "something" in the provided slice (len 0)`)
-// 	})
-
-// 	t.Run("non-empty collection but cannot match name", func(t *testing.T) {
-// 		verify(t,
-// 			[]corev1.Container{
-// 				{Name: consts.ActiveGateContainerName},
-// 				{Name: consts.StatsdContainerName},
-// 			},
-// 			consts.EecContainerName,
-// 			fmt.Sprintf(`Cannot find container "%s" in the provided slice (len 2)`, consts.EecContainerName),
-// 		)
-// 	})
-
-// 	t.Run("happy path", func(t *testing.T) {
-// 		verify(t,
-// 			[]corev1.Container{
-// 				{Name: consts.StatsdContainerName},
-// 			},
-// 			consts.StatsdContainerName,
-// 			"",
-// 		)
-// 	})
-// }
+import (
+	"context"
+	"testing"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/capability"
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/activegate/internal/authtoken"
+	"github.com/Dynatrace/dynatrace-operator/src/kubesystem"
+	"github.com/Dynatrace/dynatrace-operator/src/scheme"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testToken    = "dt.testtoken.test"
+	testUID      = "test-uid"
+	testDynakube = "test-dynakube"
+)
+
+type MockStatefulSetReconciler struct {
+	mock.Mock
+}
+
+func (m *MockStatefulSetReconciler) Reconcile() error {
+	args := m.Called()
+	return args.Error(0)
+}
+
+type MockCustomPropertiesReconciler struct {
+	mock.Mock
+}
+
+func (m *MockCustomPropertiesReconciler) Reconcile() error {
+	args := m.Called()
+	return args.Error(0)
+}
```

Comment:
you don't need to create 2 specific mocks, you could just create 1 general one, and in the test create 2 different instance of it
also you could have this general mock next (in the same package) to the actual interface definition so it can be reused in the future